src/connectors/aws_asset_ingest.py

```diff
@@ -0,0 +1,357 @@
+    landing_table = f'data.{table_name}'
+    client_id = options["client_id"]
+    secret_key = options["secret_key"]
+
+    regions = boto3.client(
+            'ec2',
+            aws_access_key_id=client_id,
+            aws_secret_access_key=secret_key
+    ).describe_regions()['Regions']
+
+    count = 0
+    if 'EC2' in table_name:
+        count = ingest_ec2(client_id, secret_key, landing_table, regions)
+    if 'SG' in table_name:
+        count = ingest_sg(client_id, secret_key, landing_table, regions)
+    if 'ELB' in table_name:
+        count = ingest_elb(client_id, secret_key, landing_table, regions)
+
+    log.info(f'Inserted {count} rows.')
+    yield count
+
+
+def ingest_ec2(client_id, secret_key, landing_table, regions):
+    instances = get_ec2_instances(client_id, secret_key, regions)
+    monitor_time = datetime.utcnow().isoformat()
+    db.insert(
+        landing_table,
+        values=[(
+            row,
+            row['InstanceId'],
+            row['Architecture'],
+            monitor_time,
+            row['InstanceType'],
+            row.get('KeyName', ''),  # can be not present if a managed instance such as EMR
+            row['LaunchTime'],
+            row['Region']['RegionName'],
+            row['State']['Name'],
+            row['InstanceName'])
+            for row in instances
+        ],
+        select='PARSE_JSON(column1), column2, column3, column4, column5, column6, column7, column8, column9, column10'
+    )
+    return len(instances)
+
+
+def ingest_sg(client_id, secret_key, landing_table, regions):
+    groups = get_all_security_groups(client_id, secret_key, regions)
+    monitor_time = datetime.utcnow().isoformat()
+    db.insert(
+        landing_table,
+        values=[(
+            row,
+            row['Description'],
+            monitor_time,
+            row['GroupId'],
+            row['GroupName'],
+            row['OwnerId'],
+            row['Region']['RegionName'],
+            row['VpcId'])
+            for row in groups],
+        select='PARSE_JSON(column1), column2, column3, column4, column5, column6, column7, column8'
+    )
+    return len(groups)
+
+
+def ingest_elb(client_id, secret_key, landing_table, regions):
+    elbs = get_all_elbs(client_id, secret_key, regions)
+    monitor_time = datetime.utcnow().isoformat()
+
+    db.insert(
+        landing_table,
+        values=[(
+            row,
+            monitor_time,
+            row['CanonicalHostedZoneName'],
+            row['CanonicalHostedZoneNameID'],
+            row['CreatedTime'],
+            row['DNSName'],
+            row['LoadBalancerName'],
+            row['Region']['RegionName'],
+            row['Scheme'],
+            row['VPCId'])
+            for row in elbs],
+        select='PARSE_JSON(column1), column2, column3, column4, column5, column6, '
+               'column7, column8, column9, column10'
+    )
+    return len(elbs)
+
+
+def get_ec2_instances(client_id, secret_key, regions):
```

Comment:
This doesn't record the information about which accounts had how many snapshots, but I have an implementation for that built in our current script and I can port it over to this connector far easier than I could explain it in a github comment, so this is fine for me.